### PR TITLE
Subgraph

### PR DESCRIPTION
**Pull request by eladmallel**

### CODE REVIEW

**Comment by eladmallel** [comment]
Example query and response:

```
query {
    streams (limit: 10) { 
        id 
        createdAt
        createdBy
        payer
        recipient
        tokenAmount
        tokenAddress
        startTime
        stopTime
        withdrawals {
            id
            withdrawnAt
            executedBy
            amount
        }
        cancellations {
            id
            cancelledAt
            cancelledBy
            payerBalance
            recipientBalance
        }
    }
}
```

response
```
{
    "data": {
        "streams": [
            {
                "id": "0xcbc26b0f296c0e3041d8b794e4d01179ccde6e2e",
                "createdAt": "1668785940",
                "createdBy": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
                "payer": "0xf39fd6e51aad88f6f4ce6ab8827279cfffb92266",
                "recipient": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
                "tokenAmount": "1000",
                "tokenAddress": "0x9fe46736679d2d9a65f0992f2272de9f3c7fa6e0",
                "startTime": "1668785455",
                "stopTime": "1668786455",
                "withdrawals": [
                    {
                        "id": "0x87c76cbe593689dd1709bf443b819c2f0bf55dbf86e93bd776a3ac2fe0532e46-1",
                        "withdrawnAt": "1668786170",
                        "executedBy": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
                        "amount": "699"
                    }
                ],
                "cancellations": [
                    {
                        "id": "0x66fead85bd928656a5777203d45e325e157e5e18c8359b03d6d822f9124166bb-1",
                        "cancelledAt": "1668786612",
                        "cancelledBy": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
                        "payerBalance": "1000",
                        "recipientBalance": "0"
                    },
                    {
                        "id": "0x7f5a811d9c51e2321f8bed9cfdf2c91be11d22168240869f3be407323e601005-0",
                        "cancelledAt": "1668786558",
                        "cancelledBy": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
                        "payerBalance": "0",
                        "recipientBalance": "0"
                    },
                    {
                        "id": "0xe3d488e9d1df28024f3fbe4790f23fef27939a99d6e2da98df218088e8a11936-2",
                        "cancelledAt": "1668786406",
                        "cancelledBy": "0x70997970c51812dc3a010c7d01b50e0d17dc79c8",
                        "payerBalance": "49",
                        "recipientBalance": "252"
                    }
                ]
            }
        ]
    }
}
```

File: script/DeployStreamAndFactory.s.sol [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.17;
+
+import "forge-std/Script.sol";
+import { Stream } from "../src/Stream.sol";
+import { StreamFactory } from "../src/StreamFactory.sol";
+
+contract DeployStreamAndFactory is Script {
```
Comment (davidbrai) [inline script/DeployStreamAndFactory.s.sol:9]: @eladmallel maybe rename to `DeployStreamFactory` ?

File: subgraph/networks.json [hunk 2]
```diff
@@ -0,0 +1,8 @@
+{
+  "mainnet": {
+    "StreamFactory": {
+      "address": "0xe7f1725E7734CE288F8367e1Bb143E90bb3F0512",
```
Comment (davidbrai) [inline subgraph/networks.json:4]: @eladmallel is this address correct?

File: script/DeployMockToken.s.sol [hunk 3]
```diff
@@ -0,0 +1,18 @@
+// SPDX-License-Identifier: GPL-3.0
+
+pragma solidity ^0.8.17;
+
+import "forge-std/Script.sol";
+import { ERC20Mock } from "openzeppelin-contracts/mocks/ERC20Mock.sol";
+
+contract DeployStreamAndFactory is Script {
```
Comment (davidbrai) [inline script/DeployMockToken.s.sol:8]: @eladmallel not sure if this file is used, but the contract name doesn't match the content

Comment (eladmallel) [reply script/DeployMockToken.s.sol:8]: you're right, will fix. it is used when testing locally, need a payment token

File: subgraph/networks.json (hunk 2 above)
Comment (eladmallel) [reply subgraph/networks.json:4]: this address is used for local testing; need to experiment further with subgraph local node config to land on a better network name

Comment (eladmallel) [reply subgraph/networks.json:4]: ok, updated the local testing network name to `local`

File: script/DeployMockToken.s.sol (hunk 3 above)
Comment (eladmallel) [reply script/DeployMockToken.s.sol]: fixed

**Review by solimander — APPROVED** [review]
Very clean 👍

**Pull request merged**